server/consumer.go

```diff
@@ -849,6 +850,9 @@ func (o *consumer) setLeader(isLeader bool) {
 		}
 		o.mu.Unlock()
 
+		// Snapshot initial info.
+		o.ici = o.info()
```

Comment:
But o.ici is captured outside of o's lock (note that initialInfo() and clearInitialInfo() do acquire lock to get o.ici).
The way I see it, o.ici should be set inside o.info() (while still under lock - see my note below in initialInfo()), or have a infoLocked() version that is assumed to run under lock (and does not have to check for o.mset), however, there may be the issue with js.clusterInfo() that requires JS lock...

Reply:
Sorry yes that was supposed to be different code path there, will fix.

Can't do infoLocked, clustering.

Reply:
Will fix.